### PR TITLE
ROC-3738: Do you need more time to respond? not available on due day

### PR DESCRIPTION
**Pull request by rlewan**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-3738

### Change description

We had a bug where the "Do you need more time to respond?" section was not available when defendant accessed the system on due day.

### Work checklist

- [X] Unit tests added where applicable

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/main/features/response/helpers/taskListBuilder.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export class TaskListBuilder {
     )
 
     const now: Moment = MomentFactory.currentDateTime()
```
Comment (geek-soft) [inline src/main/features/response/helpers/taskListBuilder.ts]: Would using `MomentFactory.currentDate` method here work the same?

File: src/main/features/response/helpers/taskListBuilder.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ export class TaskListBuilder {
...
-    if (claim.responseDeadline.isAfter(now)) {
+    if (claim.responseDeadline.endOf('day').isAfter(now)) {
```
Comment (geek-soft) [inline src/main/features/response/helpers/taskListBuilder.ts]: On the second thought in the frontend we say:

`'You’ll have until 4pm on ' + claim.responseDeadline.format('LL') + ' to respond'`

Can you ask business whether we to use end of day or 4pm as deadline?

Comment (rlewan) [reply src/main/features/response/helpers/taskListBuilder.ts]: Will do that now.

Comment (rlewan) [reply src/main/features/response/helpers/taskListBuilder.ts]: It has to be 4 PM @geek-soft, very good spot. I also need to check if CCJ is working correctly with regard to this.

File: src/main/features/response/helpers/taskListBuilder.ts [hunk 3]
```diff
@@ -11,9 +11,10 @@ import { YourDetails } from 'response/tasks/yourDetails'
 import { FreeMediationTask } from 'response/tasks/freeMediationTask'
 import { Claim } from 'claims/models/claim'
 import { WhenDidYouPayTask } from 'response/tasks/whenDidYouPayTask'
+import { isPastResponseDeadline } from 'claims/isPastResponseDeadline'
 
 export class TaskListBuilder {
-  static buildBeforeYouStartSection (draft: ResponseDraft, claim: Claim): TaskList {
+  static buildBeforeYouStartSection (draft: ResponseDraft, claim: Claim, now: moment.Moment): TaskList {
```
Comment (robertparkinson) [inline src/main/features/response/helpers/taskListBuilder.ts]: why is this being passed in? Should it not just be set inside the method?

Comment (rlewan) [reply src/main/features/response/helpers/taskListBuilder.ts]: Without it testing becomes very hard.

Comment (robertparkinson) [reply src/main/features/response/helpers/taskListBuilder.ts]: fair enough

**Pull request merged**